ledger/inventory/mem/mod.go

```diff
@@ -0,0 +1,229 @@
+package mem
+
+import (
+	"fmt"
+	"io"
+
+	"github.com/golang/protobuf/proto"
+	"go.dedis.ch/fabric/crypto"
+	"go.dedis.ch/fabric/ledger/inventory"
+	"golang.org/x/xerrors"
+)
+
+const (
+	digestLength = 32
+)
+
+// Digest is the type of ID for a page.
+type Digest [digestLength]byte
+
+func (d Digest) String() string {
+	return fmt.Sprintf("%#x", d[:4])
+}
+
+// InMemoryInventory is an implementation of the inventory interface by using a
+// memory storage which means that it will not persist.
+type InMemoryInventory struct {
+	hashFactory  crypto.HashFactory
+	pages        []inMemoryPage
+	stagingPages map[Digest]inMemoryPage
+}
+
+// NewInventory returns a new empty instance of the inventory.
+func NewInventory() *InMemoryInventory {
+	return &InMemoryInventory{
+		hashFactory: crypto.NewSha256Factory(),
+		// TODO: first page should come from the genesis block.
+		pages:        []inMemoryPage{{}},
+		stagingPages: make(map[Digest]inMemoryPage),
+	}
+}
+
+// GetPage returns the snapshot for the version if it exists, otherwise an
+// error.
+func (inv *InMemoryInventory) GetPage(index uint64) (inventory.Page, error) {
+	i := int(index)
+	if i >= len(inv.pages) {
+		return inMemoryPage{}, xerrors.Errorf("invalid page at index %d", i)
+	}
+
+	return inv.pages[i], nil
+}
+
+// Stage starts a new version. It returns the new snapshot that is not yet
+// committed to the available versions.
+func (inv *InMemoryInventory) Stage(f func(inventory.WritablePage) error) (inventory.Page, error) {
+	var page inMemoryPage
+	if len(inv.pages) > 0 {
+		// Clone the previous page of the inventory so that previous instances
+		// are carried over.
+		page = inv.pages[len(inv.pages)-1].clone()
+		page.index++
+	} else {
+		page.instances = make(map[Digest]inMemoryInstance)
+	}
+
+	err := f(page)
+	if err != nil {
+		return page, xerrors.Errorf("couldn't fill new page: %v", err)
+	}
+
+	page.root, err = page.computeHash(inv.hashFactory)
+	if err != nil {
+		return page, xerrors.Errorf("couldn't compute page hash: %v", err)
+	}
+
+	inv.stagingPages[page.root] = page
+
+	return page, nil
+}
+
+// Commit stores the snapshot with the given root permanently to the list of
```

Comment:
```suggestion
// Commit stores the snapshot with the given id permanently to the list of
```

Reply:
I renamed the parameter to be more consistent.